lib/run_loop/app.rb

```diff
@@ -92,9 +116,73 @@ def sha1
 
     private
 
+    # @!visibility private
     def plist_buddy
       @plist_buddy ||= RunLoop::PlistBuddy.new
     end
 
+    # @!visibility private
+    # An otool factory.
+    def otool(file)
+      RunLoop::Otool.new(file)
+    end
+
+    # @!visibility private
+    # A strings factory
+    def strings(file)
+      RunLoop::Strings.new(file)
+    end
+
+    # @!visibility private
+    def skip_executable_check?(file)
+      image?(file) ||
+        plist?(file) ||
+        lproj_asset?(file) ||
+        code_signing_asset?(file) ||
+        core_data_asset?(file)
```

Comment:
@jmoody Why not use a whitelist approach instead? Files that need to be checked for calabash linkage should be just the main app binary and any dylibs